.github/workflows/codeql-analysis.yml

```diff
@@ -34,15 +34,25 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v3
 
+    - run: sed -En 's/^go (.*)$/GO_VERSION=\1/p' go.mod >> $GITHUB_ENV
+      if: ${{ matrix.language == 'go' }}
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.GO_VERSION }}
+      if: ${{ matrix.language == 'go' }}
+
     - name: Initialize CodeQL
       uses: github/codeql-action/init@v2
       with:
         languages: ${{ matrix.language }}
+        debug: true
+        trap-caching: true
 
     - name: Autobuild
       uses: github/codeql-action/autobuild@v2
 
     - name: Perform CodeQL Analysis
       uses: github/codeql-action/analyze@v2
-      env:
-        CODEQL_EXTRACTOR_GO_MAX_GOROUTINES: 16
```

Comment:
Why was this removed? This came directly from GitHub tech support. However, we'll be swapping to `CODEQL_EXTRACTOR_GO_BUILD_TRACING`